src/Neogma.ts

```diff
@@ -75,4 +83,44 @@ export class Neogma {
   ): Promise<T> => {
     return getRunnable<T>(runInExisting, callback, this.driver);
   };
+
+  public generateModelFromMetadata = (
+    metadata: NeogmaModelMetadata,
+  ): NeogmaModel<any, any, any, any> => {
+    const parsedMetadata = parseModelMetadata(metadata);
+    const relations = parsedMetadata.relationships;
+    if (relations) {
+      for (const relation in relations) {
+        const relatedModel = metadata.relations[relation].model;
+        if (relatedModel !== 'self') {
```

Comment:
Same here, to save some indentation I'd use

```
      if (relatedModel === 'self') {
        continue;
      }
```

and remove this check

Reply:
LGTM
Done